src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    /// <summary>
+    /// Context for requests handled by <see cref="IRequestHandler"/>
+    /// </summary>
+    internal struct RequestContext
+    {
+        /// <summary>
+        /// The client capabilities for the request.
+        /// </summary>
+        public ClientCapabilities ClientCapabilities { get; }
+
+        /// <summary>
+        /// The LSP client making the request
+        /// </summary>
+        public string? ClientName { get; }
+
+        /// <summary>
+        /// A cancellation token
```

Comment:
Not sure if this is already planned as part of the clean up you mentioned, but maybe something more descriptive here might be helpful, or maybe even removing the comment?

Reply:
This was a straight copy and paste from the parameter, but since the PR isn't going anywhere for a little while I think I've got some time to improve it :)